client/components/LearnerViewDirective.js

```diff
@@ -21,104 +21,172 @@ tie.directive('learnerView', [function() {
+                <p ng-repeat="paragraph in instructions">
+                  {{paragraph}}
+                </p>
+              </div>
+            </div>
+            <button ng-click="showNextPrompt()" ng-if="nextButtonIsShown"
+                id="next-button">Next</button>
+            </button>
+          </div>
+          <div id="coding-ui">
+            <div id="feedback-window">
+              <div id="tie-feedback" class="tie-feedback">
+                <p ng-repeat="paragraph in feedbackMessages track by $index"
+                    class="tie-feedback-paragraph">
+                  {{paragraph}}
+                </p>
+              </div>
+            </div>
+            <div id="coding-window">
+              <div id="lang-terminal">
+                <!--textarea name="textarea" id="coding-terminal"
+                    class="terminal">
+                </textarea-->
+                <div id="coding-terminal" class="terminal">
+                  <ui-codemirror ui-codemirror="codeMirrorOptions" ng-model="code" class="codemirror-container"></ui-codemirror>
+                </div>
+                <select id="lang-select-menu" name="lang-select-menu">
+                  <option value="Python" selected>Python</option>
+                  <option value="JavaScript">JavaScript</option>
+                </select>
+                <button id="run-button" class="run-button"
+                    ng-class="{'active': !nextButtonIsShown}"
+                    ng-click="submitCode(code)"
+                    ng-disabled="nextButtonIsShown">
+                  Run
+                </button>
+              </div>
+            </div>
           </div>
-        </div>
-        <div class="tie-instructions">
-          <p ng-repeat="paragraph in instructions">
-            {{paragraph}}
-          </p>
-        </div>
-
-        <button ng-if="nextButtonIsShown" class="tie-action-button active"
-                ng-click="showNextPrompt()">
-          Next
-        </button>
-      </div>
-
-      <div class="tie-learner-view-right-column">
-        <div class="tie-coding-window">
-          <ui-codemirror ui-codemirror="codeMirrorOptions" ng-model="code"></ui-codemirror>
-        </div>
-
-        <button type="button" class="tie-action-button"
-                ng-class="{'active': !nextButtonIsShown}"
-                ng-click="submitCode(code)"
-                ng-disabled="nextButtonIsShown">
-          Run
-        </button>
-
-        <div class="tie-feedback">
-          <p ng-repeat="paragraph in feedbackMessages track by $index"
-             class="tie-feedback-paragraph">
-            {{paragraph}}
-          </p>
         </div>
       </div>
-
       <style>
-        .tie-learner-view-left-column {
+        body {
+          font-family: Roboto, 'Helvetica Neue', 'Lucida Grande', sans-serif;
+          font-size: 15px;
+          background-color: rgb(242, 242, 242);
+        }
+        #exercise-ui-outer {
```

Comment:
As a general rule, I recommend using classes rather than IDs, similar to what's already been done in this file. At the moment we have a single HTML file displaying the whole page so it doesn't matter so much, but when we start pulling things out into separate files it may be harder to maintain the requirement of IDs being unique within the entire page. Classes are more easily combined with one another, too.

Also, please prefix tie-specific classes with `.tie-`  -- that helps us distinguish classes for stuff we've written ourselves from classes we need to modify from external libs.